cdap-ui/app/features/tracker/templates/partial/add-preferred-tags-popover.html

```diff
@@ -35,15 +38,38 @@ <h3 class="modal-title">
       <div>
         <span class="fa fa-cloud-upload"
               aria-hidden="true"></span>
-        <div>Drag N Drop or<br />
-          <a href="#">Browse</a> CSV, TXT
+        <div>Drag &amp; Drop or<br />
+          <a class="file-import-link" ng-click="AddTags.importTags()">Browse</a> CSV, TXT
         </div>
+        <my-file-select class="sr-only" id="file-select" data-button-icon="fa-upload" on-file-select="AddTags.importFiles($files)" data-button-label="Import">
+        </my-file-select>
       </div>
     </div>
   </div>
-  <div class="btn-row pull-right">
-    <a href="" class="btn btn-default"
-       ng-click="$close()">Cancel</a>
-    <a href="" class="btn btn-success">Next</a>
+  <div ng-if="AddTags.proceedToNextStep === true"
+      class="text-left">
+    <div class="valid-tags">
+      <p>{{ AddTags.tagList.validTags.length }} tags loaded.</p>
+      <span ng-repeat="(key, value) in AddTags.tagList.validTags track by $index"
+            class="tag preferred">{{ value }}</a>
+    </div>
+    <div ng-if="AddTags.tagList.invalidTags.length"
+        class="invalid-tags">
+      <p class="text-danger">{{ AddTags.tagList.invalidTags.length }} tags failed to load.</p>
+      <span ng-repeat="(key, value) in AddTags.tagList.invalidTags track by $index">{{ value }}</span>
+    </div>
   </div>
 </div>
+<div class="modal-footer text-right">
+  <p class="pull-left text-danger"
+    ng-if="AddTags.proceedToNextStep === false && AddTags.tagList.invalidTags.length">Oops! Invalid format.</p>
+  <button class="btn btn-default"
```

Comment:
We might want to get the message that we display here confirmed from @nitinmotgi